src/main/java/org/bytedeco/javacv/FFmpegFrameRecorder.java

```diff
@@ -917,9 +917,7 @@ public boolean recordImage(int width, int height, int depth, int channels, int s
                passing the same picture again */
         } else {
             int step = stride * Math.abs(depth) / 8;
-            BytePointer data = image[0] instanceof ByteBuffer
-                    ? new BytePointer((ByteBuffer)image[0].position(0))
-                    : new BytePointer(new Pointer(image[0].position(0)));
+            BytePointer data = new BytePointer(new Pointer(image[0])).position(0);
```

Comment:
Leave the branching like it was so it also works with array-based ByteBuffer, at least.